### PR TITLE
update importlib_metadata dependency

### PR DESCRIPTION
**Pull request by fhoehle**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by fhoehle** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -19,11 +19,11 @@ requirements:
   host:
     - pip
     - python >=3.6
-    - setuptools_scm
+    - setuptools-scm
   run:
     - attrs >=17.4.0
     # TODO: upstream, this is conditional to <3.8, but probably not worth the non-noarch build
-    - importlib_metadata
+    - importlib-metadata
```
Comment (ocefpaf) [inline recipe/meta.yaml:26]: This does not change anything in the package really. The `_` packages there aliases to the `-`.
That is you motivation for this change?

Comment (fhoehle) [reply recipe/meta.yaml:26]: No, functionality of the package is not changed. But if you export/pin environments you will obtain duplicated packages. 
```
...
  - importlib-metadata=2.0.0=py_1
  - importlib_metadata=2.0.0=1
...
```
What is the benefit of using the alias, which was the former package name? It is very helpful package name transitions indeed.

**Comment by maresb** [comment]
It seems that @fhoehle's suggestions have been implemented, so I think this can be closed now

**Pull request closed without merging**